dashboard/app/views/devise/registrations/edit.html.haml

```diff
@@ -60,7 +60,7 @@
   -# Don't allow editing user type unless we can also edit email, because
   -# changing from a student (encrypted email) to a teacher (plaintext email)
   -# requires entering an email address.
-  - if f.object.can_edit_email?
+  - if f.object.can_edit_email? && Section.where(user_id: f.object.id).empty?
```

Comment:
Not sure (re: bad state). Since it seems useful to let users "fix" their own account, I'm modifying the implementation to only prevent the teacher to student transition.